component/registry/registry.go

```diff
@@ -0,0 +1,63 @@
+package registry
```

Comment:
`registry` has a specific meaning in docker. I would recommend having a `components` package, with registration tools.